presto-main/src/main/java/com/facebook/presto/execution/buffer/ArbitraryOutputBuffer.java

```diff
@@ -73,6 +73,9 @@
     @GuardedBy("this")
     private final ConcurrentMap<OutputBufferId, ClientBuffer> buffers = new ConcurrentHashMap<>();
 
+    //  The index of the first client buffer that should be polled
+    private final AtomicInteger nextClientBufferIndex = new AtomicInteger(0);
```

Comment:
Given the current usage you don't need nextClientBufferIndex to be an AtomicLong. volatile would suffice.

Reply:
I tend to prefer Aomic* over volatile for maintainability because of how easy it is for future changes to treat volatile fields as “regular int” and do something like use it in a loop counter. I’m fine changing it to volatile if you prefer.

Reply:
I don't have a strong preference in this case. You can leave it as is.